python/src/ignition/gazebo/helper_system.hh

```diff
@@ -0,0 +1,159 @@
+#include "ignition/gazebo/EntityComponentManager.hh"
+#include "ignition/gazebo/Export.hh"
+#include "ignition/gazebo/Server.hh"
+#include "ignition/gazebo/ServerConfig.hh"
+
+#include "destroyable.hh"
+#include "entity_component_manager.hh"
+
+namespace py = pybind11;
+
+namespace ignition
+{
+namespace gazebo
+{
+namespace python
+{
+  class HelperFixturePrivate;
+
+  class HelperFixture : public ignition::utils::python::Destroyable, public std::enable_shared_from_this<HelperFixture>
+  {
+    /// \brief Constructor
+    /// \param[in] _path Path to SDF file.
+    public: explicit HelperFixture(const std::string &_path);
+
+    /// \brief Constructor
+    /// \param[in] _config Server config file
+    public: explicit HelperFixture(const ServerConfig &_config);
+
+    /// \brief Destructor
+    public: virtual ~HelperFixture();
+
+    /// Force an early destruction of this object
+    void
+    destroy() override;
+
+    /// \brief Wrapper around a system's pre-update callback
+    /// \param[in] _cb Function to be called every pre-update
+    /// The _entity and _sdf will correspond to the world entity.
+    /// \return Reference to self.
+    public: HelperFixture &OnConfigure(std::function<void(
+        const Entity &_entity,
+        ignition::gazebo::python::EntityComponentManager &_ecm)> _cb);
+
+    /// \brief Wrapper around a system's pre-update callback
+    /// \param[in] _cb Function to be called every pre-update
+    /// \return Reference to self.
+    public: HelperFixture &OnPreUpdate(std::function<void(
+        const UpdateInfo &, ignition::gazebo::python::EntityComponentManager &)> _cb);
+
+    /// \brief Wrapper around a system's update callback
+    /// \param[in] _cb Function to be called every update
+    /// \return Reference to self.
+    public: HelperFixture &OnUpdate(std::function<void(
+        const UpdateInfo &, ignition::gazebo::python::EntityComponentManager &)> _cb);
+
+    /// \brief Wrapper around a system's post-update callback
+    /// \param[in] _cb Function to be called every post-update
+    /// \return Reference to self.
+    public: HelperFixture &OnPostUpdate(std::function<void(
+        const UpdateInfo &, const ignition::gazebo::python::EntityComponentManager &)> _cb);
+
+    /// \brief Finalize all the functions and add fixture to server.
+    /// Finalize must be called before running the server, otherwise none of the
+    /// `On*` functions will be called.
+    /// The `OnConfigure` callback is called immediately on finalize.
+    public: HelperFixture &Finalize();
+
+    /// \brief Get pointer to underlying server.
+    public: std::shared_ptr<ignition::gazebo::Server> Server() const;
+
+    /// \internal
+    /// \brief Pointer to private data.
+    // TODO(chapulina) Use IGN_UTILS_UNIQUE_IMPL_PTR(dataPtr) when porting to v6
+    private: std::shared_ptr<HelperFixturePrivate> dataPtr;
+  };
+
+/// \brief System that is inserted into the simulation loop to observe the ECM.
+class HelperSystem :
```

Comment:
This looks very similar to the `HelperSystem` here:

https://github.com/ignitionrobotics/ign-gazebo/blob/96ed1efff86af0ce5d3322fdad1bf3196896e881/src/TestFixture.cc#L27

Is there a need to duplicate it? Maybe we just need to install the other one?

Reply:
it's slightly different.

I?m using the `ignition::gazebo::python::EntityComponentManager` not the `ignition::gazebo::EntityComponentManager`

Reply:
Now that the `EntityComponentManager` is wrapped differently, do you think we can remove the duplication here too? Could we just use `TestFixture`?